### ISSUE TITLE
Store list and scalar variables in same namespace

### ISSUE THREAD

**Issue by pekkaklarck** [robotframework/robotframework#1905]
In RF 2.8.x and earlier you can create list and scalar variables with same base name and that will create too distinct variables:

```robotframework
*** Variables ***
${EXAMPLE}    Scalar variable
@{EXAMPLE}    List    variable
```

It is, however, possible to use list variables as scalars since RF 2.0.3 (#117) and scalar variables as lists since RF 2.8 (#483). Both usages require that there is only either scalar or list variable with same base name.

This situation is pretty confusing and can cause weird errors. The forthcoming addition of new dict variable (#1450) is also going to make the situation much worse. We have thus made a decision to store scalar, list and dict variables in same namespace starting from RF 2.9.

This change is going to cause backwards incompatibility problems in cases where scalar and list variables with same base name are used. That is very unfortunate, but I don't see any other option to get this implemented. We tried to deprecate this functionality already in RF 2.5 (#484), but deprecation was later removed because some of the warnings were incorrect.

**Comment by pekkaklarck** [comment]
This change is currently being done in shared-store branch with 3aeed8086ee2ca73ce2095c8c1025f238cbf4ab3 as the first commit. Tests are passing and functionality done (except for some TODOs to be resolved later) after 13ac2b8dfe730c4b116794711c2eb34d97c01c15.

**Comment by pekkaklarck** [comment]
This has been implemented. Issue open only because docs aren't updated yet.
